unipop-core/src/org/unipop/schema/property/MultiFieldPropertySchema.java

```diff
@@ -1,40 +1,43 @@
 package org.unipop.schema.property;
 
+import org.apache.commons.lang.NotImplementedException;
 import org.apache.tinkerpop.gremlin.process.traversal.P;
 import org.apache.tinkerpop.gremlin.process.traversal.step.util.HasContainer;
+import org.json.JSONArray;
+import org.json.JSONObject;
 import org.unipop.query.predicates.PredicatesHolder;
 import org.unipop.query.predicates.PredicatesHolderFactory;
 
 import java.util.*;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-
+/**
+ * Created by sbarzilay on 7/28/16.
+ */
 public class MultiFieldPropertySchema implements PropertySchema {
```

Comment:
Have you considered creating ParentSchemaProperty as a base class for this and ConcatenateFieldPropertySchema?